src/hdmf/utils.py

```diff
@@ -225,7 +224,7 @@ def __parse_args(validator, args, kwargs, enforce_type=True, enforce_shape=True,
                 ret[argname] = args[argsi]
                 argsi += 1
             else:
-                ret[argname] = arg['default']
+                ret[argname] = _copy.deepcopy(arg['default'])
```

Comment:
This is one of the key changes -- only default values are deep-copied, and only when they are used.